lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -322,6 +348,53 @@ hdPassphrase (Key rootXPub) = Passphrase $
     (unXPub rootXPub)
     ("address-hashing" :: ByteString)
 
+toAddress
+    :: Key 'RootK XPub
+    -> Key 'AddressK XPub
+    -> Index 'Hardened 'AccountK
+    -> Index 'Soft 'AddressK
+    -> Address
+toAddress rootXPub (Key addrXPub) accIx addrIx =
+    Address $ encodeBase58 bitcoinAlphabet
+    $ CBOR.toStrictByteString
+    $ encodeAddress addrXPub encodeAddressAttributes
+  where
+    encodeAddressAttributes = mempty
+        <> CBOR.encodeMapLen 1
+        <> CBOR.encodeWord8 1
+        <> encodeDerivationPath rootXPub accIx addrIx
+
+    encodeAddress :: XPub -> CBOR.Encoding -> CBOR.Encoding
+    encodeAddress (XPub pub (ChainCode cc)) encodeAttributes =
+        let
+            payload = CBOR.toStrictByteString $ mempty
+                <> CBOR.encodeListLen 3
+                <> CBOR.encodeBytes root
+                <> encodeAttributes
+                <> CBOR.encodeWord8 0
+
+            root = BA.convert $ hash @_ @Blake2b_224 $ hash @_ @SHA3_256 $ CBOR.toStrictByteString $ mempty
+                <> CBOR.encodeListLen 3
+                <> CBOR.encodeWord8 0
+                <> encodeSpendingData
+                <> encodeAttributes
+
+            encodeXPub =
+                CBOR.encodeBytes (pub <> cc)
+
+            encodeSpendingData = CBOR.encodeListLen 2
+                <> CBOR.encodeWord8 0
+                <> encodeXPub
+
+            encodeAddressPayload :: ByteString -> CBOR.Encoding
+            encodeAddressPayload pay = mempty
+                <> CBOR.encodeListLen 2
+                <> CBOR.encodeTag 24
+                <> CBOR.encodeBytes payload
+                <> CBOR.encodeWord32 (crc32 pay)
```

Comment:
This seems redundant with:

https://github.com/input-output-hk/cardano-wallet/blob/master/lib/http-bridge/src/Cardano/Wallet/HttpBridge/Binary.hs#L478-L498